packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+    Embrace.client?.metadata.userIdentifier = nil
+    resolve(true)
+  }
+  
+  @objc
+  func clearUsername(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userName = nil
+    resolve(true)
+  }
+  
+  
+  @objc
+  func endSession(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.endCurrentSession()
+    resolve(true)
+  }
+  
+  @objc
+  func checkAndSetCodePushBundleURL(
+    _ resolve: @escaping RCTPromiseResolveBlock,
+    rejecter reject: @escaping RCTPromiseRejectBlock
+  ) {
+    do {
+#if canImport(CodePush)
+      guard let url = try CodePush.bundleURL() else {
+        reject("GET_CODEPUSH_BUNDLE_URL", "Error getting Codepush Bundle URL", nil)
+        return
+      }
+      
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.javaScriptBundleURL.rawValue, value: url.path)
+      resolve(true)
+      
+#else
+      resolve(false)
+#endif
+    } catch let error {
+      reject("CHECK_AND_SET_CODEPUSH_BUNDLE_URL", "Error setting Codepush Bundle URL", error)
+    }
+  }
+  
+  @objc(addUserPersona:resolver:rejecter:)
+  func addUserPersona(_ persona: String, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.add(persona: PersonaTag(persona), lifespan: .session)
+      resolve(true)
+    } catch let error {
+      reject("ADD_USER_PERSOMAS", "Error adding an User Personas", error)
+    }
+  }
+  
+  @objc(clearUserPersonas:resolver:rejecter:)
+  func clearUserPersonas(_ persona: String, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.remove(persona: PersonaTag(persona), lifespan: .session)
+      resolve(true)
+    } catch let error {
+      reject("CLEAR_USER_PERSOMAS", "Error clearing an User Persona", error)
+    }
+  }
+  
+  @objc
+  func clearAllUserPersonas(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.removeAllPersonas()
+      resolve(true)
+    } catch let error {
+      reject("CLEAR_ALL_USER_PERSOMAS", "Error clearing all User Personas", error)
+    }
+  }
+  
+  @objc
+  func setReactNativeVersion(_ version: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.reactNativeVersion.rawValue, value: version)
+      resolve(true)
+    } catch let error {
+      reject("SET_RECT_NATIVE_VERSION", "Error setting React Native Number", error)
+    }
+  }
+  
+  @objc
+  func removeSessionProperty(_ key: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      // TODO REfactor to include lifespan
```

Comment:
this comment still relevant?

Reply:
Yes, since we need to refactor to include the other lifespan. Today we only support 1